sys/fuchsia/fidl_mappings.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package fuchsia
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// layer indicates at which layer a FidlLibrary lives in the Fuchsia build
+// system.
+type layer int
+
+const (
+	_ layer = iota
+	zircon
+	garnet
+)
+
+// FidlLibrary describes a FIDL library. It captures required details such as
+// build location, header generation, etc.
+type FidlLibrary struct {
+	layer
+
+	// FqName stores the fully-qualified name of the library in parts, e.g.
+	// the `fuchsia.mem` library is `fuchsia`, `mem`.
+	FqName []string
+}
+
+// AllFidlLibraries lists all FIDL libraries.
+var AllFidlLibraries []FidlLibrary
+
+func init() {
+	add := func(layer layer, parts ...string) {
+		AllFidlLibraries = append(AllFidlLibraries, FidlLibrary{
+			layer:  layer,
+			FqName: parts,
+		})
+	}
+	add(zircon, "fuchsia", "mem")
```

Comment:
A more idiomatic and shorter and less indirection would be to initialize the array directly. Go is expressive enough for this. If the motivation is to reduce code by replacing []string with add, then a ctor function would be more appropriate. See:
https://play.golang.org/p/isup5c9O2be

Reply:
Done

Reply:
You have not yet pushed the changes, right?
Please say PTAL when it's ready for the next round of review.

Reply:
Correct. Was using the wrong UI. Batching up now.